google_project/variables.tf

```
@@ -0,0 +1,85 @@
+variable "app_code" {
+  default     = ""
+  description = "Defaults to project_name. Used for labels and metadata on application-related resources."
+  type        = string
+}
+
+variable "billing_account_id" {
+  description = "Associated billing account"
+  type        = string
+}
+
+variable "component_code" {
+  default     = ""
+  description = "Defaults to app_code-uncat"
+}
+
+variable "cost_center" {
+  description = "Cost center of the project or resource. Default is 5650 (Services Engineering)"
+  default     = "5650"
+  type        = string
+}
+
+variable "display_name" {
+  default     = ""
+  description = "Display name for the project. Defaults to project_name-realm"
+  type        = string
+}
+
+variable "env_code" {
+  default     = ""
+  description = "Defaults to realm (e.g. environment). Used for labels and metadata on environment-related resources."
+  type        = string
+}
+
+variable "parent_id" {
+  description = "Parent folder (with GCP)."
+  type        = string
+}
+
+variable "project_name" {
+  description = "Name of project e.g., autopush"
+  type        = string
+}
+
+variable "project_id" {
+  default     = ""
+  description = "Override default project id. Only use if the project id is already taken."
+  type        = string
+}
+
+variable "realm" {
+  description = "Project environment e.g., prod"
+  type        = string
+
+  validation {
+    condition     = contains(["dev", "global", "prod", "sandbox", "stage"], var.realm)
```

```suggestion
    condition     = contains(["global", "nonprod", "prod", "sandbox"], var.realm)
```
I don't think we want dev and stage realms, they would fall under nonprod in terms of projects, such that, all nonprod resources can be in one project.

---

so, this isn't what we do in web SRE. is the idea that everyone would fall into prod / nonprod now? and does nonprod or prod include sandbox?

---

I'd like to understand the use cases/requirements around having separate projects for nonprod resources. Ideally we would just have two projects, nonprod and prod for each application/product.

I don't envision SRE using sandbox realm much, but it would have been useful for use cases when engineering teams use our module to provision 'sandbox' projects. https://github.com/mozilla-services/data-sandbox-terraform, which contains data engineering sandbox project resources, but instead, we were calling it something along the lines of 'dev' in the  project name and realm = nonprod.

---

I looked through this again, and I realized that we are setting `environment = var.realm` for gcp project labels, which I don't think we should. We should set environment labels on the actual resources (if labels are supported), e.g. set environment label on dev and stage cloud sql instances when they are both in the nonprod project.

---

makes sense to me, will remove

---

removed realm /environment, figuring it'll be setup on terraform projects specifically

---

removed all environment specific metadata from the project label.

---

Whoops sorry, I was not clear here. We need to keep realm, but environment is good to remove.